src/Http/Controllers/QuestionAdminApiController.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+namespace EscolaLms\Questionnaire\Http\Controllers;
+
+use EscolaLms\Core\Http\Controllers\EscolaLmsBaseController;
+use EscolaLms\Questionnaire\Http\Controllers\Contracts\QuestionAdminApiContract;
+use EscolaLms\Questionnaire\Http\Requests\QuestionCreateRequest;
+use EscolaLms\Questionnaire\Http\Requests\QuestionDeleteRequest;
+use EscolaLms\Questionnaire\Http\Requests\QuestionListingRequest;
+use EscolaLms\Questionnaire\Http\Requests\QuestionReadRequest;
+use EscolaLms\Questionnaire\Http\Requests\QuestionUpdateRequest;
+use EscolaLms\Questionnaire\Http\Resources\QuestionResource;
+use EscolaLms\Questionnaire\Models\Question;
+use EscolaLms\Questionnaire\Repository\Contracts\QuestionRepositoryContract;
+use Illuminate\Contracts\Support\Renderable;
+use Illuminate\Http\JsonResponse;
+use Illuminate\Support\Facades\Auth;
+
+class QuestionAdminApiController extends EscolaLmsBaseController implements QuestionAdminApiContract
+{
+    private QuestionRepositoryContract $questionRepository;
+
+    public function __construct(QuestionRepositoryContract $questionRepository)
+    {
+        $this->questionRepository = $questionRepository;
+    }
+
+    public function list(QuestionListingRequest $request): JsonResponse
+    {
+        try {
+            $questions = $this->questionRepository->searchAndPaginate();
+            return $this->sendResponseForResource(
+                QuestionResource::collection($questions),
+                "Question list retrieved successfully"
+            );
+        } catch (Renderable $e) {
+            return $this->sendError($e->getMessage());
+        }
+    }
+
+    public function create(QuestionCreateRequest $request): JsonResponse
+    {
+        try {
+            /** @var Question $question */
+            $question = Question::factory()->newModel([
```

Comment:
What are the benefits of using factory to store model in database?

Reply:
done